tests/lib/retry.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+snap_install() {
+    local PARAMS=$@
+    local CONDITIONS='timeout \| connection refused'
+    local RETRIES=2
+    local SLEEP=2
+
+    local ITER=0
+    until [ $ITER -gt $RETRIES ]; do
+        ERROR=$(snap install $PARAMS 2>&1> /dev/null)
+        if [ -z "$ERROR" ]; then 
+            break
+        elif [[ $( echo $ERROR | grep -c "$CONDITIONS") -ge 1 ]]; then
+            ITER=$[$ITER+1]
+            sleep $SLEEP
+            echo $ERROR >&2
+            if [ $ITER -le $RETRIES ]; then
+                echo "Retrying install $ITER" >&2
+            fi
+        else
```

Comment:
I don't think so because I am changing the if implementation to echo the stdout so we can use this from expect files, so both will do different things. I'll try to improve this but not sure if will be possible.